compiler/il/OMRBlock.cpp

```diff
@@ -1165,9 +1165,12 @@ static TR::SymbolReference * createSymRefForNode(TR::Compilation *comp, TR::Reso
                {
                while (valueChild->getOpCode().isArrayRef())
                   valueChild = valueChild->getFirstChild();
-
-               if (valueChild->getOpCode().isLoadVarDirect() &&
-                     valueChild->getSymbolReference()->getSymbol()->isAuto())
+               // If the node we are uncommoning is internal pointer and while iterating a child we found a node that is not 
+               // Array Reference, There are only two possibilities.
+               // 1. It is  internal poiter which was commoned means could be stored into register or on temp slot. 
+               // 2. Itself is a pointer to array object.
+               if ((valueChild->getOpCode().isLoadVarDirect() && valueChild->getSymbolReference()->getSymbol()->isAuto()) ||
+                     (valueChild->getOpCode().isLoadReg() && valueChild->getSymbolReference() && valueChild->getSymbolReference()->getSymbol()->castToAutoSymbol()->isInternalPointer()))
```

Comment:
Why `&& valueChild->getSymbolReference() &&`?  It's already been dereferenced in the line before `valueChild->getSymbolReference()->getSymbol()->isAuto()`.

This would be much clearer if some fo the repeatedly dereferenced calls were pulled into locals and reused.

Reply:
> This would be much clearer if some fo the repeatedly dereferenced calls were pulled into locals and reused.
I have made changes to use a local variable to hold value of symbol reference. 
> Why `&& valueChild->getSymbolReference()&& `? It's already been dereferenced in the line before `valueChild->getSymbolReference()->getSymbol()->isAuto()`.
Reason why we need to check that is second case (`regLoad`) here would not be `LoadVarDirect` so in that case we would not have de-referenced the symbol reference. And We can have a `regLoad` without symbol reference if it is created by Block API to split blocks after Global Register Allocation phase, and current function is actually creating one, so to make sure we do not de-reference `NULL` , I have second check there.